mRemoteV1/Config/Serializers/ConnectionSerializers/Xml/XmlConnectionsDeserializer.cs

```diff
@@ -531,6 +531,10 @@ private ConnectionInfo GetConnectionInfoFromXml(XmlNode xmlnode)
                     connectionInfo.RDPAlertIdleTimeout = xmlnode.GetAttributeAsBool("RDPAlertIdleTimeout");
                     connectionInfo.Inheritance.RDPAlertIdleTimeout =
                         xmlnode.GetAttributeAsBool("InheritRDPAlertIdleTimeout");
+					
+                    connectionInfo.VMId = xmlnode.Attributes["VMId"].Value;
+					connectionInfo.EnhancedSession = bool.Parse(xmlnode.Attributes["EnhancedSession"].Value);
+                    connectionInfo.Inheritance.EnhancedSession = bool.Parse(xmlnode.Attributes["InheritEnhancedSession"].Value);
```

Comment:
These lines should use the `GetAttributeAs...()` extension methods to safely retrieve the attribute values.

```csharp
connectionInfo.VMId = xmlnode.GetAttributeAsString("VMId");
connectionInfo.EnhancedSession = xmlnode.GetAttributeAsBool("EnhancedSession");
connectionInfo.Inheritance.EnhancedSession = xmlnode.GetAttributeAsBool("InheritEnhancedSession");
```

Reply:
Fixed by 39233d2